client/asset/dcr/dcr.go

```diff
@@ -211,95 +210,102 @@ func NewWallet(ctx context.Context, cfg *WalletConfig, logger dex.Logger, networ
 	if cfg.Account == "" {
 		cfg.Account = defaultAccountName
 	}
-	dcr := unconnectedWallet(ctx, cfg, logger)
-	notifications := &rpcclient.NotificationHandlers{
-		OnBlockConnected:  dcr.onBlockConnected,
-		OnClientConnected: dcr.onClientConnected,
-	}
-	dcr.client, err = connectNodeRPC(walletCfg.RPCListen, walletCfg.RPCUser, walletCfg.RPCPass,
-		walletCfg.RPCCert, notifications)
-	if err != nil {
-		return nil, err
-	}
-	// Check the min api versions.
-	versions, err := dcr.client.Version()
-	if err != nil {
-		return nil, fmt.Errorf("error retrieving API versions: %v", err)
-	}
-	err = checkVersionInfo(versions)
-	if err != nil {
-		return nil, err
-	}
-	// Subscribe to block updates.
-	err = dcr.client.NotifyBlocks()
+	dcr := unconnectedWallet(cfg, logger)
+
+	dcr.client, err = newClient(walletCfg.RPCListen, walletCfg.RPCUser,
+		walletCfg.RPCPass, walletCfg.RPCCert)
 	if err != nil {
-		return nil, fmt.Errorf("error registering for block notifications")
+		return nil, fmt.Errorf("DCR ExchangeWallet.Run error: %v", err)
 	}
 	// Beyond this point, only node
 	dcr.node = dcr.client
+
 	return dcr, nil
 }
 
 // unconnectedWallet returns an ExchangeWallet without a node. The node should
 // be set before use.
-func unconnectedWallet(ctx context.Context, cfg *WalletConfig, logger dex.Logger) *ExchangeWallet {
-	dcr := &ExchangeWallet{
-		ctx:         ctx,
+func unconnectedWallet(cfg *asset.WalletConfig, logger dex.Logger) *ExchangeWallet {
+	return &ExchangeWallet{
 		log:         logger,
 		acct:        cfg.Account,
 		tradeChange: make(map[string]time.Time),
-		nfo:         cfg.AssetInfo,
 		tipChange:   cfg.TipChange,
 	}
-	go dcr.monitorConnection()
-	return dcr
 }
 
-// connectNodeRPC attempts to create a new websocket connection to a dcrwallet
+// newClient attempts to create a new websocket connection to a dcrwallet
 // instance with the given credentials and notification handlers.
-func connectNodeRPC(host, user, pass, cert string,
-	notifications *rpcclient.NotificationHandlers) (*rpcclient.Client, error) {
+func newClient(host, user, pass, cert string) (*rpcclient.Client, error) {
 
 	dcrdCerts, err := ioutil.ReadFile(cert)
 	if err != nil {
 		return nil, fmt.Errorf("TLS certificate read error: %v", err)
 	}
 
 	config := &rpcclient.ConnConfig{
-		Host:         host,
-		Endpoint:     "ws",
-		User:         user,
-		Pass:         pass,
-		Certificates: dcrdCerts,
+		Host:                host,
+		Endpoint:            "ws",
+		User:                user,
+		Pass:                pass,
+		Certificates:        dcrdCerts,
+		DisableConnectOnNew: true,
 	}
 
-	dcrdClient, err := rpcclient.New(config, notifications)
+	dcrdClient, err := rpcclient.New(config, nil)
```

Comment:
The docs and variable names are a bit confusing since they mix the use of dcrwallet and dcrd, when really it's either.  Perhaps a generic term like RPC server.